### PR TITLE
remove some redundant fields

### PR DESCRIPTION
**Pull request by SimonCropp**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5453)

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs [hunk 1]
```diff
@@ -80,11 +78,7 @@ public TemplateJintPreprocessor(IResourceFileReader resourceCollection, Resource
         {
             if (!string.IsNullOrWhiteSpace(scriptResource.Content))
             {
-                _engine = SetupEngine(resourceCollection, scriptResource, context);
-            }
-            else
-            {
-                _engine = null;
+                SetupEngine(resourceCollection, scriptResource, context);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs]: `_engine = ...`

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs [hunk 2]
```diff
@@ -14,13 +14,11 @@ namespace Microsoft.DocAsCode.Build.TableOfContents
 
     internal sealed class TocResolver
     {
-        private readonly IHostService _host;
         private readonly Dictionary<string, TocItemInfo> _collection;
         private readonly Dictionary<FileAndType, TocItemInfo> _notInProjectTocCache = new Dictionary<FileAndType, TocItemInfo>();
 
-        public TocResolver(IHostService host, Dictionary<string, TocItemInfo> collection)
+        public TocResolver(Dictionary<string, TocItemInfo> collection)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs]: DocFx tries to provide backward compatibility for public APIs. For this change, we can mark `public TocResolver(IHostService host, Dictionary<string, TocItemInfo> collection) : this(collection) { }` as `[Obsolete]`, and add another constructor removing the unneeded parameter.

Same for other public API signature changes.

Comment (SimonCropp) [reply src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs]: but that class is internal

File: src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs (hunk 1 above)
Comment (SimonCropp) [reply src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs]: @superyyrrzz is that necessary? it doest seem _engine  is used?

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs (hunk 2 above)
Comment (SimonCropp) [reply src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs]: what are the other public APIs i changed?

File: src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs (hunk 1 above)
Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.Engine/TemplateProcessors/Preprocessors/TemplateJintPreprocessor.cs]: You are right... Just notice this.

**Pull request merged**